src/sentry/migrations/0572_rm_cficache_symcache.py

```diff
@@ -0,0 +1,55 @@
+# Generated by Django 3.2.20 on 2023-10-05 09:51
+
+from django.db import migrations
+
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+# These two filetypes have not been created ever since
+# <https://github.com/getsentry/sentry/commit/c14c8f0e3a30fb5c2221fdefa4c0f5167bdc8986#diff-08e824e030b14e7936ffeda41ccf81d8feed2cdcdfbc522d0c46d0064f3640c9L750>
+# But there seems to be quite some leftover `File` objects sticking around we want to remove.
+TYPES_TO_DELETE = ("project.symcache", "project.cficache")
+
+
+def rm_cficache_symcache(apps, schema_editor):
+    File = apps.get_model("sentry", "File")
+
+    # According to <https://develop.sentry.dev/database-migrations/#filters>:
+    # > it is better to iterate over the entire table instead of using a filter
+    # Also, there is no index on `type` anyway.
+    for file in RangeQuerySetWrapperWithProgressBar(File.objects.all()):
```

Comment:
Have you considered to push the file type filter to the database? It still requires a full table scan, but the operation should be significantly faster since fewer results need to be streamed to python.

```suggestion
    for file in RangeQuerySetWrapperWithProgressBar(File.objects.filter(type__in=TYPES_TO_DELETE)):
```

Reply:
The table scan can make the query slow enough that we hit query timeouts.